tests/ui/regex.rs

```diff
@@ -118,7 +118,33 @@ fn trivial_regex() {
     let _ = BRegex::new(r"\b{start}word\b{end}");
 }
 
+fn regex_creation_in_loops() {
+    loop {
+        let regex = Regex::new("a.b");
+        //~^ ERROR: compiling a regex in a loop
+        let regex = BRegex::new("a.b");
+        //~^ ERROR: compiling a regex in a loop
+        #[allow(clippy::regex_creation_in_loops)]
+        let allowed_regex = Regex::new("a.b");
+
+        if true {
+            let regex = Regex::new("a.b");
+            //~^ ERROR: compiling a regex in a loop
+        }
+
+        for _ in 0..10 {
+            let nested_regex = Regex::new("a.b");
+            //~^ ERROR: compiling a regex in a loop
+        }
+    }
+
+    for i in 0..10 {
+        let dependant_regex = Regex::new(&format!("{i}"));
+    }
+}
```

Comment:
One more thing: I think we currently lint for `Regex::new()` call in a static if it's in a loop, can you add a test for it?
```rs
loop {
  static R: LazyLock<Regex> = LazyLock::new(|| Regex::new("a.b").unwrap());
}
```

We could store the loop expr's HirId alongside the Span and compare the `hir_id.owner` of the loop and the call to check they're in the same item.

Reply:
Done, good catch!